scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -864,9 +864,106 @@
       (log/warn "Skipping a subset of matches because of rate-limit:" matches-throttled))
     matches-kept))
 
+(defn handle-launch-task-metrics
+  [matches task-txns pool-name]
+  (let [offers-matched (->> matches
+                            (mapcat :leases)
+                            (map :offer))
+        num-offers-matched (->> offers-matched
+                                (map (comp :value :id))
+                                distinct
+                                count)
+        user->num-jobs (->> matches
+                            (mapcat :task-metadata-seq)
+                            (map (comp cached-queries/job-ent->user :job :task-request))
+                            frequencies)]
+    (log-structured/info "Launching matched tasks"
+                         {:number-offers-matched num-offers-matched
+                          :number-tasks (count task-txns)
+                          :number-jobs-per-user (print-str user->num-jobs) ; don't treat the map as json
+                          :pool pool-name})
+    (meters/mark! scheduler-offer-matched num-offers-matched)
+    (histograms/update! number-offers-matched num-offers-matched))
+  (meters/mark! (meters/meter (metric-title "matched-tasks" pool-name)) (count task-txns)))
+
 (def kill-lock-timer-for-launch (timers/timer ["cook-mesos" "scheduler" "kill-lock-acquire-for-launch"]))
 
+(defn launch-tasks-for-cluster
```

Comment:
I believe I'll need to change this to 
1. grab the lock for the cc, 
2. write this cc's matches to the db and then 
3. launch for this cc
4. release lock in the finally
Instead of:
1. grab global lock
2. write all matches
3. run launch threads for each cc
4. release lock in the finally

Reply:
Instead of:
1. grab global lock
2. write all matches
3. run launch threads for each cc
4. release lock in the finally